backend/controllers/usersControllers.js

```diff
@@ -66,6 +68,63 @@ module.exports.register = async (req, res) => {
   };
 
 
+  const transporter = nodemailer.createTransport({
+    host: 'smtp.gmail.com',
+    port: 465,
+    secure: true,
+    auth: {
+      user: process.env.EMAIL,
+      pass: process.env.PASSWORD,
+    },
+  });
+
+module.exports.forgotPassword = async (req, res) => {
+    const { email } = req.body;
+    const user = await User.findOne({ email });
+  
+    if (!user) {
+      return res.status(400).json({ message: 'User with this email does not exist.' });
+    }
+  
+    const otp = crypto.randomBytes(3).toString('hex');
+    const otpExpires = Date.now() + 3600000; // 1 hour
+  
+    user.otp = otp;
+    user.otpExpires = otpExpires;
+    await user.save();
+  
+    const mailOptions = {
+      from: process.env.EMAIL,
+      to: user.email,
+      subject: 'Password Reset OTP',
+      text: `Your OTP for password reset is: ${otp}`
+    };
+  
+    transporter.sendMail(mailOptions, (error, info) => {
+      if (error) {
+        console.error('Error sending email:', error);
```

Comment:
Here, please send some details about the error, in the response body also, so that the user can identify the mistake.